### PR TITLE
RavenDB-14418 Enhancements for PatternForOutputReduceToCollectionReferences

### PR DESCRIPTION
**Pull request by arekpalinski**
https://issues.hibernatingrhinos.com/issue/RavenDB-14418

### CODE REVIEW

**Review by ppekrol — APPROVED** [review]
LGTM, but requires Studio work.

File: src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs [hunk 1]
```diff
@@ -1760,8 +1760,21 @@ protected override Expression VisitMethodCall(MethodCallExpression node)
             {
                 var type = node.Method.GetGenericArguments()[0];
                 _loadDocumentTypes.Add(type);
-                var collection = _conventions.GetCollectionName(type);
-                Out($", \"{collection}\"");
+
+                var parameters = node.Method.GetParameters();
+                if (parameters.Length == 1)
+                {
+                    var collection = _conventions.GetCollectionName(type);
```
Comment (ayende) [inline src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs:1767]: Do we need to do escaping here?

**Review by ayende — APPROVED** [review]

File: src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs:1767]: I'm not sure I follow. Did you mean one line below:
`Out($", \"{collection}\"");`
? 
We have `\"` to pass collection name as string parameter - `LoadDocument("users/1", "Users")`

**Comment by arekpalinski** [comment]
@Danielle9897 will take care of the studio part in this PR

File: src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs:1767]: Yes, what happens if the collection contains `"` ?

Comment (arekpalinski) [reply src/Raven.Client/Documents/Indexes/ExpressionStringBuilder.cs:1767]: If `FindCollectionName` convention was overwritten and returned collection name would include special characters then I think yes we would need to escape it here.

I opened issue to verify all usages: https://issues.hibernatingrhinos.com/issue/RavenDB-14681

**Pull request merged**